session106/XMap/XMap/Model/TFLFetcher.swift

```diff
@@ -57,19 +63,19 @@ class TFLFetcher: NSObject {
         executor.execute { (responseJSON) in
             
             // Received Data for parsing
-            // Do some work...Mahyad?
+            guard let tubes = responseJSON as? Array<Any> else {return}
+            
+            var parsedTubes = [Tube]()
             
-            do {
-                let decoder = JSONDecoder()
-                let arrivalTube = try decoder.decode(ArrivalTube.self, from: responseJSON as! Data)
-                print(arrivalTube)
-            } catch {
-                print("fuck me")
+            for tube in tubes {
+                if let tubeData = try? JSONSerialization.data(withJSONObject: tube, options: []), let arrivalTube = try? JSONDecoder().decode(Tube.self, from: tubeData) {
+                    parsedTubes.append(arrivalTube)
+                }
             }
-//            print(responseJSON)
+            
             DispatchQueue.main.async {
-                handler?(responseJSON)
-            }
+                handler?(parsedTubes)
```

Comment:
Ditto. The handler: `TFLFetcherArrivalHandler` should not be `(Any) -> ()`, it should be `([Tube]) -> ()`